app/Commands/UploadTrackCommand.php

```diff
@@ -136,9 +136,15 @@ public function execute()
                 $trackFile = new TrackFile();
                 $trackFile->is_master = $name === 'FLAC' ? true : false;
                 $trackFile->format = $name;
+                if (array_key_exists($name, Track::$CacheableFormats) && $trackFile->is_master == false) {
+                    $trackFile->is_cacheable = true;
+                } else {
+                    $trackFile->is_cacheable = false;
+                }
                 $track->trackFiles()->save($trackFile);
 
-                $target = $destination . '/' . $trackFile->getFilename(); //$track->getFilenameFor($name);
+                // Encode track file
+                $target = $destination . '/' . $trackFile->getFilename();
```

Comment:
Use the `getFile()` method on `TrackFile` objects to get an absolute path to the file - `getFilename()` should only be used for when the file's own name, and nothing else, is needed.